pykickstart/commands/syspurpose.py

```diff
@@ -0,0 +1,82 @@
+# copy, or redistribute it subject to the terms and conditions of the GNU
+# General Public License v.2.  This program is distributed in the hope that it
+# will be useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  Any Red Hat
+# trademarks that are incorporated in the source code or documentation are not
+# subject to the GNU General Public License and may only be used or replicated
+# with the express permission of Red Hat, Inc.
+#
+
+from pykickstart.base import BaseData, KickstartCommand
+from pykickstart.errors import KickstartParseError
+from pykickstart.options import KSOptionParser
+from pykickstart.version import F29
+
+from pykickstart.i18n import _
+
+class F29_SystemPurpose(KickstartCommand):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        KickstartCommand.__init__(self, writePriority, *args, **kwargs)
+        self.op = self._getParser()
+
+        self.role = kwargs.get("role", None)
+        self.sla = kwargs.get("sla", None)
+        self.usage = kwargs.get("usage", None)
+
+    def __str__(self):
+        retval = KickstartCommand.__str__(self)
+
+        if not retval and not any([self.role, self.sla, self.usage]):
+            return ""
+
+        retval += "# Intended system purpose\nsyspurpose"
+        if self.role:
+            retval+=" --role=%s" % self.role
+        if self.sla:
+            retval+=" --sla=%s" % self.sla
+        if self.usage:
+            retval+=" --usage=%s" % self.usage
+
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="syspurpose", description="""
+                            The syspurpose command is used to describe
+                            how the system is intended to be used after the installation.
+
+                            This information then can be used to apply the
+                            correct subscription entitlement to the system.""", version=F29)
+        op.add_argument("--role", metavar="<role_name>", version=F29, required=False,
+                        help="""
+                        The intended role of the system.""")
+        op.add_argument("--sla", metavar="<sla_name>", version=F29, required=False,
+                        help="""
+                        Name of the sla intended for the system.""")
+        op.add_argument("--usage", metavar="<usage>", version=F29, required=False,
+                        help="""
+                        The intended usage of the system.""")
```

Comment:
Could you expand what you mean by role, SLA (service level agreement?), and usage in the help just a little bit?  I'm not sure what any of them means in this context.  Just a sentence or so would be sufficient.